ui/src/Login.tsx

```diff
@@ -44,7 +44,7 @@ function Login() {
         })
         .then((res) => {
           setCookie("userToken", res.headers["authorization"], { path: "/" });
-          let token = cookies.userToken.replace("Bearer ", "");
+          let token = res.headers["authorization"].replace("Bearer ", "");
```

Comment:
This line change on my local allowed me to login immediately instead of twice. It does not fix the fact that the cookie isn't set until the user changes pages, but I believe this is okay since when the user successfully logs in, they are rerouted to a new page and the cookie is set there